lib/math.py

```diff
@@ -0,0 +1,162 @@
+#NOTE: This is ugly.
+from __go__.math import (Pi, E, Ceil, Copysign, Abs, Floor, Mod, Frexp, IsInf,
+    IsNaN, Exp2, Modf, Trunc, Exp, Expm1, Log, Log1p, Log10, Pow, Sqrt, Acos,
+    Asin, Atan, Atan2, Hypot, Sin, Cos, Tan, Acosh, Asinh, Atanh, Sinh, Cosh,
+    Tanh, Erf, Erfc, Gamma, Lgamma) # pylint: disable=g-multiple-import
+
+# Constants
+
+pi = Pi
+e = E
+
+# Number-theoretic and representation functions
+
+def ceil(x):
+    return Ceil(x)
+    
+def copysign(x,y):
+    return Copysign(x,y)
+    
+def fabs(x):
```

Comment:
Inputs should be converted to floats so that the following works:

```
>>> class A(object):
...   def __float__(self):
...     return 3.14
... 
>>> math.fabs(A())
3.14
```

Applies to most functions in this file.